aws/resource_aws_cloudwatch_dashboard.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloudWatchDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudWatchDashboardPut,
+		Read:   resourceAwsCloudWatchDashboardRead,
+		Update: resourceAwsCloudWatchDashboardPut,
+		Delete: resourceAwsCloudWatchDashboardDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		// Note that we specify both the `dashboard_body` and
+		// the `dashboard_name` as being required, even though
+		// according to the REST API documentation both are
+		// optional: http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutDashboard.html#API_PutDashboard_RequestParameters
+		Schema: map[string]*schema.Schema{
+			"dashboard_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard_body": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
+				},
+				DiffSuppressFunc: suppressEquivalentJsonDiffs,
+			},
+			"dashboard_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
I'm not sure that `dashboard_body` and `dashboard_name` should be set to `ForceNew`. According to the terraform documentation, that implies changes to those resources should mean that a new resource should be created? However, they should be update-able (even though this code treats them as required, they're not required as per the AWS docs, meaning we shouldn't force the creation of new resources due to changes in these fields).